src/UnityMvvmToolkit.Core/Internal/ObjectHandlers/ObjectWrapperHandler.cs

```diff
@@ -56,7 +56,7 @@ public ObjectWrapperHandler(ValueConverterHandler valueConverterHandler)
             var targetType = typeof(TValueType);
             var contextProperty = GetMemberValue(context, memberInfo, out var sourceType);
 
-            if (targetType == sourceType)
+            if (targetType == sourceType && String.IsNullOrEmpty(bindingData.ConverterName))
```

Comment:
It will lead to the cast exception if the property is read-only. I am working on a solution. Will be published soon.